CodeStarterCamp_Week2/PlayLotto.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  PlayLotto.swift
+//  CodeStarterCamp_Week2
+//
+//  Created by 정선아 on 2022/04/22.
+//
+
+import Foundation
+
+func playLotto() {
+    pickLottoNumbers()
+    matchNumber(myLottoNumbers, lottoNumbers: lottoNumbers)
+    comfirmNumbers()
+}
+
+//찍은 번호
+let myLottoNumbers: [Int] = [4, 13, 24, 36, 38, 45]
+
+//로또 번호 배열
+var lottoNumbers: [Int] = []
+
+//로또 번호 랜덤 뽑기
+func pickLottoNumbers() {
+    while lottoNumbers.count < 6 {
+        let number = Int.random(in: 1...45)
+        if lottoNumbers.contains(number) == false {
+            lottoNumbers.append(number)
+        }
+    }
+}
+
+//찍은 번호와 로또 번호가 일치하면 들어가는 배열
+var matchedNumbers: [Int] = []
+
+func matchNumber(_ myLottoNumbers: [Int], lottoNumbers: [Int]) {
+    for number in myLottoNumbers {
+        if lottoNumbers.contains(number) {
+            matchedNumbers.append(number)
+        }
+    }
+}
+
+//matchedNumbers를 String타입으로 변환하는 과정
+func convertArray(from array: [Int]) -> [String] {
+    let convertArray = array.map({ (number: Int) -> String in return "\(number)"
+        return "\(number)"
+    })
+    return convertArray
+}
+
+let matchedNumbersToString = convertArray(from: matchedNumbers)
+
+//결과 출력
+func comfirmNumbers() {
+    if matchedNumbers.isEmpty {
+        print("아쉽지만 겹치는 번호가 없습니다.")
+    } else {
+        print("축하합니다! 겹치는 번호는 \(matchedNumbersToString.joined(separator: ",")) 입니다!")
+    }
```

Comment:
위에서 메서드를 고쳤다면, 여기는 아래와 같이 `문자열 보간법` 안에 메서드를 호출하면서 바로 변환 결과를 내보내는 식으로 처리하면 됩니다. 😄 

```suggestion
    } else {
        print("축하합니다! 겹치는 번호는 \(convertToString(from: matchedNumbers)) 입니다!")
    }
```

문자열 보간법 부분이 너무 코드가 긴 것 같아요. 😢 
라는 생각이 드신다면...
`임시 상수`로 담아주면서 이게 "메시지" 라는 걸 강조해주고, 문자열 보간법은 깔끔하게 가져가는 방법도 있습니다.

```swift
let message = convertToString(from: matchedNumbers)
print("축하합니다! 겹치는 번호는 \(message) 입니다!")
```

Reply:
와우!!!! 저런 방법이 있을 수도 있겠군요!
저도 ```print 함수``` 출력되는 부분이 너무 복잡하다 싶었는데... 혹시 제 마음 속을 들여다보시는 건 아닌가요??
덕분에 문자열 보간법을 깔끔하게 만들었어요:)

Reply:
ㅎㅎ 비교를 위해, 보간법 안에 직접적으로 메서드를 호출해봤어요! (일부로요)
근데 솜도 느끼셨겠지만, 저렇게 사용하면 가독성 별로죠. 😅 

한 번 임시 상수로 받아준 뒤에, 깔끔한 표현으로 넣어주는 게 센스 있는 방법입니다. 👍🏻